packages/thrift-server-core/src/main/parse_json.ts

```diff
@@ -0,0 +1,257 @@
+ * https://raw.githubusercontent.com/apache/thrift/master/lib/nodejs/lib/thrift/json_parse.js
+ */
+
+import { Int64 } from './Int64'
+
+type JsonValue = {} | Array<any> | string | number | boolean | null | undefined
+
+class JsonParser {
+    private at = 0
+    private ch = ''
+    private escapee: Record<string, string> = {
+        '"': '"',
+        '\\': '\\',
+        '/': '/',
+        'b': '\b',
+        'f': '\f',
+        'n': '\n',
+        'r': '\r',
+        't': '\t',
+    }
+    private text = ''
+
+    constructor(source: string) {
+        this.text = source
+        this.at = 0
+        this.ch = ' '
+    }
+
+    public parse(): any {
+        const result = this.value()
+        this.white()
+        if (this.getCh()) {
+            throw new SyntaxError('Syntax error')
+        }
+
+        return result
+    }
+
+    private array() {
+        // Parse an array value.
+        const array: Array<any> = []
+
+        if (this.getCh() === '[') {
+            this.next('[')
+            this.white()
+            if (this.getCh() === ']') {
+                this.next(']')
+                return array // empty array
+            }
+            while (this.getCh()) {
+                array.push(this.value())
+                this.white()
+                if (this.getCh() === ']') {
+                    this.next(']')
+                    return array
+                }
+                this.next(',')
+                this.white()
+            }
+        }
+        throw new SyntaxError('Bad array')
+    }
+
+    private object() {
+        // Parse an object value.
+
+        let key: string | undefined = ''
+        const object: Record<string, JsonValue> = {}
+
+        if (this.getCh() === '{') {
+            this.next('{')
+            this.white()
+            if (this.getCh() === '}') {
+                this.next('}')
+                return object // empty object
+            }
+            while (this.getCh()) {
+                key = this.string()
+                this.white()
+                this.next(':')
+                if (Object.hasOwnProperty.call(object, key)) {
+                    throw new SyntaxError('Duplicate key "' + key + '"')
+                }
+                object[key] = this.value()
+                this.white()
+                if (this.getCh() === '}') {
+                    this.next('}')
+                    return object
+                }
+                this.next(',')
+                this.white()
+            }
+        }
+        throw new SyntaxError('Bad object')
+    }
+
+    private value(): JsonValue {
+        // Parse a JSON value. It could be an object, an array, a string, a number,
+        // or a word.
+
+        this.white()
+        switch (this.getCh()) {
+            case '{':
+                return this.object()
+            case '[':
+                return this.array()
+            case '"':
+                return this.string()
+            case '-':
```

Comment:
not sure if this can be an issue.. how about value being .1 instead of 0.1 ... this shall not parse it as number..

Reply:
Really nice catch... This is consistent with Apache (https://github.com/apache/thrift/blob/master/lib/nodejs/lib/thrift/json_parse.js#L260) as it's mostly copied over. We should fix this though.

Reply:
https://github.com/creditkarma/thrift-server/pull/99 fixes that